Fix prev/next links on public profile page

valrus opened the pull request:
This fixes #4950 by adding a "prev" link on the public profile page after the "next" link is clicked.

I changed the logic a bit as well. Previously the "next" link would show up whenever there were any statuses on the current page, leading to a "There's nothing here" page if there were no further-back statuses to retrieve. Now it doesn't show up once you've hit the end. Both new behaviors are demonstrated in this screenshot:
![pagination_fix](https://user-images.githubusercontent.com/343665/36348974-d96b281e-1430-11e8-817e-b0c0078fe9a5.jpg)

This is my first PR! 🎉  I'm happy to scale back my fix if the changes beyond what was strictly necessary to fix #4950 are deemed inappropriate here, and please let me know if this is a case that would benefit from a test! I like writing tests when possible but in my experience they're not usually that helpful for UI stuff like this.

ykzts commented on app/controllers/accounts_controller.rb:
```diff
@@ -18,7 +18,8 @@ def show
         @statuses        = cache_collection(@statuses, Status)
-        @next_url        = next_url unless @statuses.empty?
+        @next_url        = next_url if @statuses.last.id > filtered_statuses.last.id
+        @prev_url        = prev_url if @statuses.first.id < filtered_statuses.first.id
```
```ruby
unless @statuses.empty
  @next_url = next_url if @statuses.last.id > filtered_statuses.last.id
  @prev_url  = prev_url if @statuses.first.id < filtered_statuses.first.id
end
```

ykzts commented on app/views/accounts/show.html.haml:
```diff
@@ -39,6 +39,9 @@
     .pagination
-      = link_to safe_join([t('pagination.next'), fa_icon('chevron-right')], ' '), @next_url, class: 'next', rel: 'next'
+      - if @prev_url
+        = link_to safe_join([t('pagination.prev'), fa_icon('chevron-left')], ' '), @prev_url, class: 'prev', rel: 'previous'
```
`safe_join([fa_icon('chevron-left'), t('pagination.prev')], ' ')`

Gargron commented on app/controllers/accounts_controller.rb:
```diff
@@ -70,12 +73,20 @@ def set_account
+  end
+
+  def prev_url
+    pagination_url(since_id: @statuses.first.id)
```
The reason there's no previous button is because since_id doesn't do what a previous button should do. since_id will display the most recent items potentially cut off by the ID, not the items *immediately before* the ID.

Cassolotl commented:
Prevous/next is a thing that causes confusion on some Tumblr layouts, and the solution is for those to be called older/newer instead. Can we do that here too?

valrus commented:
@Cassolotl: Absolutely, if and when I figure out how to address the _crippling flaw_ pointed out by @Gargron.

The solutions I've thought of so far seem inelegant but I do still intend to implement something here.

valrus commented:
Finally got around to a second pass at this, which works around the problem that `since_id` doesn't do what we want. I made a new pagination scope for this, with a comment explaining the purpose and distinction.

As well, I moved to "older/newer" per @Cassolotl's request. There were no localization strings for those so I added them just for English, but if an automatic translation would suffice I can add them for other languages too. (If there are any tools for adding localizations, please let me know.)

With the move to "older"/"newer" I thought it made more sense to have "older" on the left and "newer" on the right. All told, it now looks like this (with page size at 3 for tidiness, but it's still 20 in the PR):
![masto_new_paging](https://user-images.githubusercontent.com/343665/36648469-d52f98e8-1a48-11e8-94a1-8c0660f48e05.jpg)
